infrastructure/cdk/src/providers/analytics.ts

```diff
@@ -2,55 +2,58 @@ import { Construct } from "constructs"
 import * as cdk from "aws-cdk-lib"
 import * as s3 from "aws-cdk-lib/aws-s3"
 import * as glue from "@aws-cdk/aws-glue-alpha"
-import { Schema, eventSchema, actionSchema } from "@casimir/data"
+import { Schema, eventSchema } from "@casimir/data"
 import { kebabCase, pascalCase, snakeCase } from "@casimir/format"
 import { Config } from "./config"
 import { AnalyticsStackProps } from "../interfaces/StackProps"
+import { CfnWorkGroup } from "aws-cdk-lib/aws-athena"
 
 /**
  * Data analytics stack
  */
 export class AnalyticsStack extends cdk.Stack {
     public readonly name = pascalCase("analytics")
+    public readonly workGroupName: string
 
     constructor(scope: Construct, id: string, props: AnalyticsStackProps) {
         super(scope, id, props)
 
         const config = new Config()
 
         const eventColumns = new Schema(eventSchema).getGlueColumns()
-        const actionColumns = new Schema(actionSchema).getGlueColumns()
 
-        const database = new glue.Database(this, config.getFullStackResourceName(this.name, "database"), {
-            databaseName: snakeCase(config.getFullStackResourceName(this.name, "database"))
+        const database = new glue.Database(this, config.getFullStackResourceName(this.name, "database", config.dataVersion), {
+            databaseName: snakeCase(config.getFullStackResourceName(this.name, "database", config.dataVersion)),
         })
 
         const eventBucket = new s3.Bucket(this, config.getFullStackResourceName(this.name, "event-bucket", config.dataVersion), {
             bucketName: kebabCase(config.getFullStackResourceName(this.name, "event-bucket", config.dataVersion))
         })
 
-        const actionBucket = new s3.Bucket(this, config.getFullStackResourceName(this.name, "action-bucket", config.dataVersion), {
-            bucketName: kebabCase(config.getFullStackResourceName(this.name, "action-bucket", config.dataVersion))
-        })
-        
-        new s3.Bucket(this, config.getFullStackResourceName(this.name, "output-bucket", config.dataVersion), {
+        const outputBucket = new s3.Bucket(this, config.getFullStackResourceName(this.name, "output-bucket", config.dataVersion), {
             bucketName: kebabCase(config.getFullStackResourceName(this.name, "output-bucket", config.dataVersion))
         })
 
+        const workGroup = new CfnWorkGroup(this, config.getFullStackResourceName(this.name, "workgroup", config.dataVersion), {
+            name: config.getFullStackResourceName(this.name, "workgroup", config.dataVersion),
+            recursiveDeleteOption: true,
+            state: "ENABLED",
+            workGroupConfiguration: {
+                resultConfiguration: {
+                    outputLocation: `s3://${outputBucket.bucketName}/`,
+                },
+            },
+            tags: [{ key: "version", value: config.dataVersion.toString() }],
+        })
+
+        this.workGroupName = workGroup.name
```

Comment:
Where are you using the workGroupName?

Is it at all necessary to tie the work group to the glue database? Or this is just on the Athena side?

Reply:
I am going to replace `expect(workgroup).toBeDefined()` with matching the workGroupName.
If we don't tie them here with a workgroup then we have to update the location in the console manually.